lib/web_ui/lib/src/engine/configuration.dart

```diff
@@ -76,16 +82,9 @@ FlutterConfiguration? _configuration;
 @visibleForTesting
 void debugOverrideJsConfiguration(JsFlutterConfiguration? newConfig) {
   if (newConfig != null) {
-    final JSObject newJsConfig = objectConstructor.assign(
-      <String, Object>{}.jsify(),
-      _jsConfiguration.jsify(),
-      newConfig.jsify(),
-    );
-    _configuration = FlutterConfiguration()
-        ..setUserConfiguration(newJsConfig as JsFlutterConfiguration);
-    print('Overridden engine JS config to: ${newJsConfig.dartify()}');
+    _debugConfiguration = configuration.withOverrides(newConfig);
```

Comment:
```suggestion
    _debugConfiguration = _configuration.withOverrides(newConfig);
```

Or do we want to allow to override the overridden values?

Reply:
No, `configuration.withOverrides` is correct. Whatever mechanism was used to configure the tests, we should use that as our base set. Right now dart2wasm tests still use `window.flutterConfiguration` and the JS tests use `flutter.js` and explicitly pass the config in, so we should support both.